src/pages/home/report/withReportAndReportActionOrNotFound.js

```diff
@@ -0,0 +1,153 @@
+        reportActions: PropTypes.shape(reportActionPropTypes),
+
+        /** The policies which the user has access to */
+        policies: PropTypes.objectOf(
+            PropTypes.shape({
+                /** The policy name */
+                name: PropTypes.string,
+
+                /** The type of the policy */
+                type: PropTypes.string,
+            }),
+        ),
+
+        /** Route params */
+        route: PropTypes.shape({
+            params: PropTypes.shape({
+                /** Report ID passed via route */
+                reportID: PropTypes.string,
+
+                /** ReportActionID passed via route */
+                reportActionID: PropTypes.string,
+            }),
+        }).isRequired,
+
+        /** Beta features list */
+        betas: PropTypes.arrayOf(PropTypes.string),
+
+        /** Indicated whether the report data is loading */
+        isLoadingReportData: PropTypes.bool,
+
+        /** Is the window width narrow, like on a mobile device? */
+        isSmallScreenWidth: PropTypes.bool.isRequired,
+    };
+
+    const defaultProps = {
+        forwardedRef: () => {},
+        reportActions: {},
+        report: {},
+        policies: {},
+        betas: [],
+        isLoadingReportData: true,
+    };
+
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportAndReportActionOrNotFound(props) {
+        // For small screen, we don't call openReport API when we go to a sub report page by deeplink
+        // So we need to call openReport here for small screen
+        useEffect(() => {
+            if (!props.isSmallScreenWidth || !_.isEmpty(props.report)) {
+                return;
+            }
+            Report.openReport(props.route.params.reportID);
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, []);
+
+        const getReportAction = useCallback(() => {
+            let reportAction = props.reportActions[`${props.route.params.reportActionID}`];
+
+            // Handle threads if needed
+            if (reportAction === undefined || reportAction.reportActionID === undefined) {
+                reportAction = ReportActionsUtils.getParentReportAction(props.report);
+            }
+
+            return reportAction;
+        }, [props.report, props.reportActions, props.route.params.reportActionID]);
+
+        const reportAction = getReportAction();
+
+        // Perform all the loading checks
+        const isLoadingReport = props.isLoadingReportData && (_.isEmpty(props.report) || !props.report.reportID);
+        const isLoadingReportAction = _.isEmpty(props.reportActions) || (props.report.isLoadingReportActions && _.isEmpty(getReportAction()));
+        const shouldHideReport = !isLoadingReport && (_.isEmpty(props.report) || !props.report.reportID || !ReportUtils.canAccessReport(props.report, props.policies, props.betas));
+
+        if ((isLoadingReport || isLoadingReportAction) && !shouldHideReport) {
```

Comment:
Inline with the above comment, anyway to remove `!shouldHideReport` from here so that the flag is only initialized after this block?

Reply:
Okay thanks.